private/model/api/codegentest/service/awsendpointdiscoverytest/endpoint_discovery_test.go

```diff
@@ -78,12 +73,20 @@ func TestAsyncEndpointDiscovery(t *testing.T) {
 			t.Errorf("expected %q, but received %q", e, a)
 		}
 	})
+	req.Handlers.Complete.PushBack(func(r *request.Request) {
+		firstAsyncReq.Done()
+	})
 	if err := req.Send(); err != nil {
 		t.Fatal(err)
 	}
 
-	firstAsyncReq.Done()
-	descWg.Wait()
+	for s := time.Now().Add(1 * time.Second); s.After(time.Now()); {
+		// Wait for the cache to be updated before making second request.
+		if svc.endpointCache.Has(req.Operation.Name) {
```

Comment:
What if this never happens? Looks to me like this will never halt, but I'm not sure how that time loop bit works. Even if it does halt after x time, what if the cache still isn't updated? Seems like you could give yourself a better error message in that case.

Reply:
Thanks, I'll add a better error message here.